### ISSUE TITLE
[DocDB] Public IP deployment "Error retrieving leader master URL" from the master UI follower in another zone

### ISSUE THREAD

**Issue by FranckPachot** [yugabyte/yugabyte-db#11603]
### Description

I'm deploying a lab with two zones across public IP between two zones:
```
zone           public IP        private IP

eu-fra-1-ad-3  yb1.pachot.net   10.0.0.231
eu-fra-1-ad-3  yb2.pachot.net   10.0.0.200

eu-zrh-1-ad-1  yb3.pachot.net   10.0.0.14

```
with the following:

```
ssh -q opc@yb1.pachot.net /home/opc/yugabyte/bin/yb-master --master_addresses=10.0.0.231:7100,yb2.pachot.net:7100,yb3.pachot.net:7100 --server_broadcast_addresses=yb1.pachot.net:7100 --rpc_bind_addresses=10.0.0.231:7100 --use_private_ip=zone --placement_cloud oci --placement_region eu-fra-1 --placement_zone eu-fra-1-ad-3 --fs_data_dirs=/home/opc/var/data --replication_factor=3 --default_memory_limit_to_ram_ratio=0.20  </dev/null &
ssh -q opc@yb2.pachot.net /home/opc/yugabyte/bin/yb-master --master_addresses=yb1.pachot.net:7100,10.0.0.200:7100,yb3.pachot.net:7100 --server_broadcast_addresses=yb2.pachot.net:7100 --rpc_bind_addresses=10.0.0.200:7100 --use_private_ip=zone --placement_cloud oci --placement_region eu-fra-1 --placement_zone eu-fra-1-ad-3 --fs_data_dirs=/home/opc/var/data --replication_factor=3 --default_memory_limit_to_ram_ratio=0.20  </dev/null &
ssh -q opc@yb3.pachot.net /home/opc/yugabyte/bin/yb-master --master_addresses=yb1.pachot.net:7100,yb2.pachot.net:7100,10.0.0.14:7100 --server_broadcast_addresses=yb3.pachot.net:7100  --rpc_bind_addresses=10.0.0.14:7100 --use_private_ip=zone --placement_cloud oci --placement_region eu-zrh-1 --placement_zone eu-zrh-1-ad-1 --fs_data_dirs=/home/opc/var/data --replication_factor=3 --default_memory_limit_to_ram_ratio=0.20   </dev/null &
```

It looks good but http://yb3.pachot.net:7000/ shows:
```
Error retrieving leader master URL: http://10.0.0.200:7000/?raw
Error: Network error (yb/util/curl_util.cc:55): curl error: Couldn't connect to server.
```
and in my opinion, this should use the public IP because the leader (http://yb2.pachot.net:7000/) is accessible though public IP only. The other follower (http://yb1.pachot.net:7000/) can show the UI because it is in the same zone as the leader.

**Comment by bmatican** [comment]
This is likely the same problem as with #7867
We should probably use the server_broadcast_address, if one is specified.

**Comment by druzac** [comment]
This should be fixed by [e99e29481e596b53d9261096457829eccc3c253f](https://github.com/yugabyte/yugabyte-db/commit/e99e29481e596b53d9261096457829eccc3c253f).
